### PR TITLE
In-place concrete translation for BallX

### PR DESCRIPTION
**Pull request by mforets**
see https://github.com/JuliaReach/LazySets.jl/pull/1995

### CODE REVIEW

File: src/Sets/Ball1.jl [hunk 1]
```diff
@@ -278,11 +278,39 @@ A translated ball in the 1-norm.
 ### Algorithm
 
 We add the vector to the center of the ball.
+
+### Notes
+
+See also [`translate!(::Ball1, AbstractVector)`](@ref) for the in-place version.
```
Comment (schillic) [inline src/Sets/Ball1.jl:284]:
```suggestion
See also [`translate!(::Ball1, ::AbstractVector)`](@ref) for the in-place version.
```
Same for the other seven docstrings.

File: src/Sets/Ball2.jl [hunk 2]
```diff
@@ -262,11 +262,39 @@ A translated ball in the 2-norm.
...
+See also [`translate!(::Ball2, AbstractVector)`](@ref) for the in-place version.
 """
 function translate(B::Ball2, v::AbstractVector)
+    return translate(copy(B), v)
+end
+
+"""
+    translate!(B::Ball2, v::AbstractVector)
+
+Translate (i.e., shift) a ball in the 1-norm by a given vector, in-place.
```
Comment (schillic) [inline src/Sets/Ball2.jl:277]:
```suggestion
Translate (i.e., shift) a ball in the 2-norm by a given vector, in-place.
```

**Review by schillic — APPROVED** [review]

File: src/Sets/Ballp.jl [hunk 3]
```diff
@@ -256,11 +256,39 @@ A translated ball in the p- norm.
 ### Algorithm
 
 We add the vector to the center of the ball.
+
+### Notes
+
+See also [`translate!(::Ballp, AbstractVector)`](@ref) for the in-place version.
 """
 function translate(B::Ballp, v::AbstractVector)
+    return translate(copy(B), v)
```
Comment (schillic) [inline src/Sets/Ballp.jl:265]:
```suggestion
    return translate!(copy(B), v)
```

File: src/Sets/BallInf.jl [hunk 4]
```diff
@@ -323,11 +323,39 @@ A translated ball in the infinity norm.
 ### Algorithm
 
 We add the vector to the center of the ball.
+
+### Notes
+
+See also [`translate!(::BallInf, AbstractVector)`](@ref) for the in-place version.
 """
 function translate(B::BallInf, v::AbstractVector)
+    return translate(copy(B), v)
```
Comment (schillic) [inline src/Sets/BallInf.jl:332]:
```suggestion
    return translate!(copy(B), v)
```

File: src/Sets/Ball2.jl [hunk 5]
```diff
@@ -262,11 +262,39 @@ A translated ball in the 2-norm.
 ### Algorithm
 
 We add the vector to the center of the ball.
+
+### Notes
+
+See also [`translate!(::Ball2, AbstractVector)`](@ref) for the in-place version.
 """
 function translate(B::Ball2, v::AbstractVector)
+    return translate(copy(B), v)
```
Comment (schillic) [inline src/Sets/Ball2.jl:271]:
```suggestion
    return translate!(copy(B), v)
```

File: src/Sets/Ball1.jl [hunk 6]
```diff
@@ -278,11 +278,39 @@ A translated ball in the 1-norm.
...
 """
 function translate(B::Ball1, v::AbstractVector)
+    return translate(copy(B), v)
```
Comment (schillic) [inline src/Sets/Ball1.jl:287]:
```suggestion
    return translate!(copy(B), v)
```

**Review by schillic — COMMENTED** [review]
Oops :smile:

File: src/Sets/Ball1.jl [hunk 7]
```diff
@@ -278,11 +278,39 @@ A translated ball in the 1-norm.
...
+- `B` -- ball in the 1-norm
+- `v` -- translation vector
+
+### Output
+
+The ball `B` translated by `v`.
+
+### Notes
+
+See also [`translate(::Ball1, AbstractVector)`](@ref) for the out-of-place version.
```
Comment (mforets) [inline src/Sets/Ball1.jl:306]:
```suggestion
See also [`translate(::Ball1, ::AbstractVector)`](@ref) for the out-of-place version.
```

File: src/Sets/Ball2.jl [hunk 8]
```diff
@@ -262,11 +262,39 @@ A translated ball in the 2-norm.
...
+- `B` -- ball in the 2-norm
+- `v` -- translation vector
+
+### Output
+
+The ball `B` translated by `v`.
+
+### Notes
+
+See also [`translate(::Ball2, AbstractVector)`](@ref) for the out-of-place version.
```
Comment (mforets) [inline src/Sets/Ball2.jl:290]:
```suggestion
See also [`translate(::Ball2, ::AbstractVector)`](@ref) for the out-of-place version.
```

File: src/Sets/BallInf.jl [hunk 9]
```diff
@@ -323,11 +323,39 @@ A translated ball in the infinity norm.
...
+- `B` -- ball in the infinity norm
+- `v` -- translation vector
+
+### Output
+
+The ball `B` translated by `v`.
+
+### Notes
+
+See also [`translate(::BallInf, AbstractVector)`](@ref) for the out-of-place version.
```
Comment (mforets) [inline src/Sets/BallInf.jl:351]:
```suggestion
See also [`translate(::BallInf, ::AbstractVector)`](@ref) for the out-of-place version.
```

File: src/Sets/Ballp.jl [hunk 10]
```diff
@@ -256,11 +256,39 @@ A translated ball in the p- norm.
 ### Algorithm
 
 We add the vector to the center of the ball.
+
+### Notes
+
+See also [`translate!(::Ballp, AbstractVector)`](@ref) for the in-place version.
```
Comment (mforets) [inline src/Sets/Ballp.jl:262]:
```suggestion
See also [`translate!(::Ballp, ::AbstractVector)`](@ref) for the in-place version.
```

File: src/Sets/Ballp.jl [hunk 11]
```diff
@@ -256,11 +256,39 @@ A translated ball in the p- norm.
...
+- `B` -- ball in the p-norm
+- `v` -- translation vector
+
+### Output
+
+The ball `B` translated by `v`.
+
+### Notes
+
+See also [`translate(::Ballp, AbstractVector)`](@ref) for the out-of-place version.
```
Comment (mforets) [inline src/Sets/Ballp.jl:284]:
```suggestion
See also [`translate(::Ballp, ::AbstractVector)`](@ref) for the out-of-place version.
```

**Review by schillic — APPROVED** [review]

**Pull request merged**